src/error.rs

```diff
@@ -196,17 +193,13 @@ impl Error {
 	pub fn source_range(&self) -> std::ops::Range<usize> {
 		let (start, len) = match &self {
 			Self::InvalidEscapeSequence(e) => {
-				if e.character.is_some() {
-					(e.position, 2)
-				} else {
-					(e.position, 1)
-				}
+				(e.position, 1 + e.character.map_or(0, |c| c.len_utf8()))
```

Comment:
I'm not sure if `c.len_utf8()` is safe, because there might be multiple ways to encode the same character? Or am I confusing this with the possibility of having multiple multi-codepoint sequences for the same grapheme?

Reply:
IIRC there's only one way to encode a codepoint in UTF-8, so this definitely won't cause panic or anything, if that's what you mean by safe.

One grapheme cluster can indeed consist of multiple codepoints, but handling that correctly would require [`unicode-segmentation`](https://github.com/unicode-rs/unicode-segmentation), which IMO is totally overkill for our use case here.

Reply:
Yeah I think you're right. There are multiple ways to encode the same grapheme, but only one way to encode a codepoint.

That is important, otherwise we could still end up slicing wrong. But I'm confident now that `char::utf8_len()` is correct here.